ambry-api/src/main/java/com.github.ambry/router/CryptoService.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+/**
+ * Crypto Service to assist in encrypting or decrypting bytes
+ */
+public interface CryptoService<T> {
```

Comment:
I see that it is required for KeyManagementService, but none of the methods in CryptoService use it right now. Are there any additional methods that will need it in the future?